rskj-core/src/main/java/co/rsk/peg/Federation.java

```diff
@@ -168,25 +152,15 @@ public boolean equals(Object other) {
         }
 
         Federation otherFederation = (Federation) other;
-
-        return this.getNumberOfSignaturesRequired() == otherFederation.getNumberOfSignaturesRequired() &&
-            this.getSize() == otherFederation.getSize() &&
-            this.getCreationTime().equals(otherFederation.getCreationTime()) &&
-            this.creationBlockNumber == otherFederation.creationBlockNumber &&
-            this.btcParams.equals(otherFederation.btcParams) &&
-            this.members.equals(otherFederation.members) &&
-            this.getRedeemScript().equals(otherFederation.getRedeemScript());
+        return this.getAddress().equals(otherFederation.getAddress());
     }
 
     @Override
     public int hashCode() {
         // Can use java.util.Objects.hash since all of Instant, int and List<BtcECKey> have
         // well-defined hashCode()s
         return Objects.hash(
```

Comment:
```suggestion
        return getAddress().hashCode();
```
Would that make sense?